### PR TITLE
Julia 1.0 support

### PR DESCRIPTION
**Pull request by crstnbr**
This PR makes Git.jl run under Julia 1.0 (issue #20). It is a continuation of #18.

* I have "issues" running the tests. They should work, but they seem to take forever (in particular `mktree`). Has this always been the case? I only quickly checked with Julia 0.6 and also aborted tests after ~10 minutes there.
* It would be nice to check the build script under linux / macos (no previous git installation). But perhaps it is fine to merge and wait for user complains. After all, this package doesn't seem to be used by many.

### CODE REVIEW

**Comment by crstnbr** [comment]
Just for myself:
Travis+Appveyor Julia 0.6: `ERROR: LoadError: function string does not accept keyword arguments`

**Comment by crstnbr** [comment]
Appveyor fail seems unrelated.

Travis fails for Julia 0.6 on MacOS: `No output has been received in the last 10m0s`.

Don't know what to do here. Everything else seems to be fine.

**Comment by crstnbr** [comment]
From my side this seems to be ready for a review @ararslan.

**Comment by ararslan** [comment]
You should be able to drop support for Julia 0.6 entirely, remove the use of Compat, and add 0.7 and 1.0 to Travis and AppVeyor testing.

File: REQUIRE [hunk 1]
```diff
@@ -1,4 +1,4 @@
 julia 0.6
```
Comment (ararslan) [inline REQUIRE]: Should be set to 0.7

Comment (crstnbr) [reply REQUIRE]: Of course.

**Comment by crstnbr** [comment]
Ready for merge?

**Comment by ararslan** [comment]
https://github.com/JuliaLang/METADATA.jl/pull/19408

**Comment by crstnbr** [comment]
Thanks for merging this quickly and tagging a new release!
Both open issues can be closed as well.

**Pull request merged**